middleware/ack.go

```diff
@@ -0,0 +1,22 @@
+package middleware
```

Comment:
I think a middleware like this is a great "compromise" which make it easy to turn of auto-ack but still acknowledge every delivery like this! Linked to my comment about the `ackAwareAcknowledger`, shouldn't at least we as creators of this handle it?

```go
type knowsIsHandled interface {
    IsHandled() bool
}

// ...
if aaa, ok := d.Acknowledger.(knowsIsHandled); ok {
    if aaa.IsHandled() {
        return
    }
}

if err := d.Ack(false); err != nil {
    fmt.Printf("could not ack delivery %s: %v", d.CorrelationId, err)
}

// ...
```